### ISSUE TITLE
WSS/HTTPS setup with qztray

### ISSUE THREAD

**Issue by adic26** [qzind/tray#400]
Hello,

I tried following down to the T : https://github.com/qzind/tray/issues/73

Yet I wasn't able to get the qztray to work with setup, however I was able to get it to work HTTP/WS.

My Setup:
QZTray - Customized build with a self-signed cert : running on a machine in a network with mappings routed in the router via port-forwarding

GCP: Running my application on google cloud using a `Lets Encrypt` cert, but the qztray client side contains the Self-Signed Cert + PrivateKey that I initially made for the print server.

The Above Works for non-http, however I cannot get it to work HTTPS despite of `cscript "%PROGRAMFILES%\QZ Tray\auth\windows-keygen.js" "%PROGRAMFILES%\QZ Tray" install "" "" "path\to\fullchain.pem" "path\to\privkey.pem"` where fullchain and privKey is `Lets Encrypt` certs with qztray client key combo for `self-signed` cert

Please advice!

**Comment by tresf** [comment]
I'm not sure what the self-signed cert has to do with it.  For SSL, the two PEM files should work and then you'll need to restart QZ Tray.

If LetsEncrypt gave you two certs, try the one without the full chain.

HTTP should always work because there's no domain information in the insecure request, just listening on port 80.

**Comment by tresf** [comment]
> The self-signed cert: I was simply following this article: [link removed]

That's the licensing component.  That has nothing to do with SSL.

> Okay I think i'll give the LetsEncrypt without the full chain a try on a brand new qztray installation 

If SSL is the issue, I strongly recommend using the vanilla installer.  We don't assist with any setups that bypass the licensing component and I can reassure you the two are unrelated.

**Comment by adic26** [comment]
> If SSL is the issue, I strongly recommend using the vanilla installer. We don't assist with any setups that bypass the licensing component and I can reassure you the two are unrelated.

Installed Vanilla QZTray app with :
`cscript "%PROGRAMFILES%\QZ Tray\auth\windows-keygen.js" "%PROGRAMFILES%\QZ Tray" install "" "" "path\to\main-crt.pem" "path\to\privkey.pem"`

Despite of this under HTTPS:

> 2019-01-16 06:07:17,151 [WARN] badMessage: 400 Illegal character 0x16 for HttpChannelOverHttp@516a6f43{r=0,c=false,a=IDLE,uri=}
> 2019-01-16 06:07:18,229 [WARN] Illegal character 0x16 in state=START for buffer HeapByteBuffer@2e1ca495[p=1,l=517,c=16384,r=516]={\x16<<<\x03\x01\x02\x00\x01\x00\x01\xFc\x03\x03\xC2\xB1j\x87\xDe\xAcg...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00>>>\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}
> 2019-01-16 06:07:18,229 [WARN] badMessage: 400 Illegal character 0x16 for HttpChannelOverHttp@4dec079f{r=0,c=false,a=IDLE,uri=}
> 2019-01-16 06:07:18,291 [WARN] Illegal character 0x16 in state=START for buffer HeapByteBuffer@2e1ca495[p=1,l=168,c=16384,r=167]={\x16<<<\x03\x01\x00\xA3\x01\x00\x00\x9f\x03\x03I\xD2\x8fB\x06\xD3\x92...\x17\x00\x18\x00\x1b\x00\x03\x02\x00\x02jj\x00\x01\x00>>>\x00\x00uP\x00\x00\x00\x0b\x00\x02\x01\x00\x003\x00+\x00...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}
> 2019-01-16 06:07:18,291 [WARN] badMessage: 400 Illegal character 0x16 for HttpChannelOverHttp@4c360df5{r=0,c=false,a=IDLE,uri=}

Please advice, we are in need of desperate help here as we are unable to print receipts from the outside server.

**Comment by tresf** [comment]
> Please advice, we are in need of desperate help here as we are unable to print receipts from the outside server.

This is a bug tracker.  This is NOT a channel for urgent support.  QZ Industries, LLC offers a support plan via https://qz.io.  If you have a premium subscription, please contact support@qz.io so the issue can be escalated accordingly.

> `400 Illegal character 0x16 for HttpChannelOverHttp`

This suggests you're trying to connect to HTTP using HTTPS.  Did you modify the `.connect()` parameters in a way which forces HTTP ports?  Does the same happen with https://demo.qz.io (you can drop to a the browser's console and type your connect command manually). By default, `qz-tray.js` will only connect to HTTPS on secure ports and HTTP on insecure ports.  If you're connecting HTTPS to an insecure port, something very strange is happening.

**Comment by adic26** [comment]
Fair enough, I should have taken this to community. Although there is a `help wanted` label which made me think that I can ask questions here.

@tresf thanks for your help. 

For people with the same issues:

Turns out secure port was being used same as insecure port. Now that's been resolved, the only thing I had to do was re-run the following command with the domain name that is associated with fullchain/private key:
`cscript "%PROGRAMFILES%\QZ Tray\auth\windows-keygen.js" "%PROGRAMFILES%\QZ Tray" install "<domain_name>" "" "path\to\main-crt.pem" "path\to\privkey.pem"`

Finally , point your `<domain_name>` DNS to print server public router address.

Thanks for your help!
